libs/core/src/components/sage-tabs/sage-tabs.tsx

```diff
@@ -0,0 +1,164 @@
+import { Component, Element, Host, h, Prop, Listen } from '@stencil/core';
+@Component({
+  tag: 'sage-tabs',
+  styleUrl: 'sage-tabs.scss',
+  shadow: true,
+  // scoped: true,
+})
+export class SageTabs {
+  private tabs;
+  private tabPanels;
+  @Element() el: HTMLDivElement;
+
+  /**
+   * Sets the aria-label attached to the tablist element, required
+  */
+  @Prop() tablistLabel: string;
+ 
+  /**
+    * Sets unique id on tabs component, required
+  */
+  @Prop() componentId: string;
+  
+  /**
+   * Sets tabs variant styles as outlined in Figma documentation, optional
+   * @defaultValue primary
+    */
+  @Prop() variant?: 'primary' | 'availability' | 'filter' = 'primary';
+
+  /**
+   * Sets default active tab, required
+   */
+  @Prop({mutable: true}) activeTab: string;
+
+  @Listen('tabClick', {
+    passive: true,
+    target: 'body',
+  }) 
+  tabClickHandler(event: CustomEvent<string>) {
+    if (this.componentId === event.detail[1]) {
+      this.activeTab = event.detail[0];
+    }
+  }
+
+  private findAllChildren = () => {
+    this.tabs = this.el.querySelectorAll('sage-tab');
+    this.tabPanels = this.el.querySelectorAll('sage-tabpanel');
+  }
+
+  private passPropsToChildren = () => {
+    this.tabs.forEach(child => {
+      child['activeTab'] = this.activeTab.toString();
+      child['parentComponent'] = this.componentId.toString();
+      child['variant'] = this.variant.toString();
+    });
+    this.tabPanels.forEach(child => {
+      child['activeTab'] = this.activeTab.toString();
+      child['parentComponent'] = this.componentId.toString();
+      child['variant'] = this.variant.toString();
+    });
+  }
+
+  @Listen('keydown', {
+    passive: true,
+  })
+  handleKeyDown(ev: KeyboardEvent){
+    const tabList = Array.from(this.el.querySelectorAll('[role="tab"]'));
+    const activeEl = this.getActiveElement();
+    const tabLocations = this.getTabLocations(tabList);
+    const firstTabNumber = 0;
+    const lastTabNumber = tabLocations.length - 1;
+
+    if (ev.key === 'ArrowLeft') {
+      if (activeEl.id === tabLocations[0].id) {
+        tabLocations[lastTabNumber].focus();
+        this.activeTab = tabLocations[lastTabNumber].id;
```

Comment:
This may not be needed. The issue I'm raising here is, do we want the tab content to open while the keyboard user traverses the tab-list, then activate their choice by clicking space. Since don't know what kind of content will exist auto loading on focus could give a stuttered experience

Reply:
The W3C best practices notes:
>It is recommended that tabs activate automatically when they receive focus as long as their associated tab panels are displayed without noticeable latency. This typically requires tab panel content to be preloaded. Otherwise, automatic activation slows focus movement, which significantly hampers users' ability to navigate efficiently across the tab list.
I'd opt to defaulting to automatic switching and iterating the component if we ever come across an issue with noticeable latency. The component is built with the ability to pass all content directly into the tabpanel on load.